### PR TITLE
Add Hume Shield API

### PR DESCRIPTION
**Pull request by Thundermaker300**
Bonus: Player.Stamina and Player.StaminaStat

### CODE REVIEW

**Review by Michal78900 — APPROVED** [review]

**Comment by Michal78900** [comment]
Fix build errors

File: Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -739,11 +739,26 @@ public float MaxArtificialHealth
             }
         }
 
+        /// <summary>
+        /// Gets or sets the player's Hume Shield.
+        /// </summary>
+        /// <remarks>This value can bypass <see cref="FpcRole.MaxHumeShield"/>. However, this value will only be visible to the end-player as Hume Shield if <see cref="FpcRole.IsHumeShieldedRole"/> is <see langword="true"/>. Otherwise, the game will treat the player as though they have the amount of Hume Shield specified, even though they cannot see it.</remarks>
+        public float HumeShield
```
Comment (iopietro) [inline Exiled.API/Features/Player.cs:746]: Define an interface for the HumeShield like the game does (`IHumeShieldedRole`) and implement it only in classes that are able to show it in the HUD. We did a similar thing with the `ISubroutinedScpRole` interface.

If you want to wrap their HumeShieldModule, you can create the "HumeShieldedRole" class which implements the IHumeShieldedRole interface, then make all role that can see the hume shield inherit from it.

File: Exiled.API/Features/Player.cs [hunk 2]
```diff
@@ -2374,6 +2399,17 @@ public bool TryGetSessionVariable<T>(string key, out T result)
...
+        /// <summary>
+        /// Plays the Hume Shield break sound effect from the player.
+        /// </summary>
+        /// <remarks>This will only function if the player's <see cref="FpcRole.IsHumeShieldedRole"/> is <see langword="true"/>.</remarks>
+        public void PlayShieldBreakSound()
+        {
+            PlayerRoles.PlayableScps.HumeShield.DynamicHumeShieldController.ShieldBreakMessage message = new();
+            message.Target = ReferenceHub;
+            message.SendToAuthenticated();
+        }
```
Comment (iopietro) [inline Exiled.API/Features/Player.cs:2411]:
```suggestion
        public void PlayShieldBreakSound() 
        => new PlayerRoles.PlayableScps.HumeShield.DynamicHumeShieldController.ShieldBreakMessage(){ Target = ReferenceHub }.SendToAuthenticated();
```

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Roles/FpcRole.cs [hunk 3]
```diff
@@ -110,6 +111,16 @@ public PlayerMovementState MoveState
...
 
+        /// <summary>
+        /// Gets a value indicating whether or not this role is protected by a hume shield.
+        /// </summary>
+        public bool IsHumeShieldedRole => this is IHumeShieldRole;
+
+        /// <summary>
+        /// Gets a value indicating the role's maximum Hume Shield. This will always be <c>0</c> if <see cref="IsHumeShieldedRole"/> is <see langword="false"/>.
+        /// </summary>
+        public float MaxHumeShield => !IsHumeShieldedRole ? 0f : (Base as IHumeShieldedRole).HumeShieldModule.HsMax;
```
Comment (iopietro) [inline Exiled.API/Features/Roles/FpcRole.cs:122]:
```suggestion
```
It's already included in the `HumeShieldBase` class, you can wrap it inside hume shield roles or just leave it as it is.

File: Exiled.API/Features/Roles/IHumeShieldRole.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+    /// <summary>
+    /// Represents a role that supports a hume shield.
+    /// </summary>
+    public interface IHumeShieldRole
+    {
+        /// <summary>
+        /// Gets a reference to the role's <see cref="HumeShieldModuleBase"/>.
+        /// </summary>
+        public HumeShieldModuleBase HumeShieldBase { get; }
```
Comment (iopietro) [inline Exiled.API/Features/Roles/IHumeShieldRole.cs:13]:
```suggestion
        HumeShieldModuleBase HumeShieldBase { get; }
```
Interface access modifiers are public by default.

**Comment by iopietro** [comment]
Fix build errors and I'll merge it.

**Pull request merged**